### DISCUSSION TITLE
Can I use Regex with filterTypes in global filter?

### DISCUSSION THREAD

**Discussion by robSturcke** [TanStack/table#3686] [Q&A] [answered]
Been having some trouble figuring out how to get the `GlobalFilter` to handle a string as well as Regex.

I looked into `filterTypes` within the docs but had trouble finding proper implementation.

[My CodeSandbox here.](https://codesandbox.io/s/react-table-global-filter-pagination-forked-wm1tw?file=/src/App.js)

My confusion is because I can use a similar function to this in the JS console and it works fine, but when I re-work it to the `filterType` within the code example I'm having issues getting it to filter both strings and regex.

What am I misinterpreting/misunderstanding from what I'm taking away from the docs?

Examples for regex to test:

`car[fe]` or `.*car.*`

```
 const filterTypes = useMemo(
    () => ({
      // Override the default text filter to use
      // "startWith"
      text: (rows, id, filterValue) => {
        return rows.filter((row) => {
          const rowValue = row.values[id];
          const isRegexMatch = () => {
            try {
              console.log(RegExp(filterValue));
              return RegExp(filterValue).test(String(rowValue));
            } catch (err) {
              return false;
            }
          };
          return rowValue !== undefined
            ? String(rowValue)
                .toLowerCase()
                .startsWith(String(filterValue).toLowerCase()) || isRegexMatch()
            : true;
        });
      },
    }),
    []
  );
```

My useTable:
```
const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    prepareRow,
    page,
    canPreviousPage,
    canNextPage,
    pageOptions,
    pageCount,
    gotoPage,
    nextPage,
    previousPage,
    setPageSize,
    preGlobalFilteredRows,
    setGlobalFilter,
    state,
    state: { pageIndex, pageSize },
  } = useTable(
    {
      columns,
      data,
      filterTypes,
    },
    useFilters,
    useGlobalFilter,
    useSortBy,
    usePagination
  );
```

Thank you for your time, I greatly appreciate it.

**Comment by robSturcke** [comment, marked as answer]
Over-analyzed the problem.  Had to import my original `regexFilter` func as `regex: regexFilter,` then declare `const globalFilter = 'regex';`

**Reply by ghost** [reply]
@robSturcke I am very new to React and trying to implement the same. Could you please share your regexFilter function here. I think this is a common scenario and its gonna help many beginners like me
